portability-gateway/src/main/java/org/dataportabilityproject/gateway/crypto/EncrypterImpl.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.dataportabilityproject.gateway.crypto;
+
+import com.google.common.base.Charsets;
+import com.google.common.io.BaseEncoding;
+import java.security.InvalidKeyException;
+import java.security.Key;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import javax.crypto.BadPaddingException;
+import javax.crypto.Cipher;
+import javax.crypto.IllegalBlockSizeException;
+import javax.crypto.NoSuchPaddingException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Provides AES and RSA-based encryption implementations. See {@link EncrypterFactory} to create.
+ */
+final class EncrypterImpl implements Encrypter {
+  private static final Logger logger = LoggerFactory.getLogger(EncrypterImpl.class);
+  private static final String SYMMETRIC_ALGORITHM = "AES";
```

Comment:
looks like you started doing it this way:

Can you have SYMMETRIC_ALGO and PUBLIC_PRIVATE_ALGO = "RSA/ECB/PKCS1Padding" and take a boolean isSymmetric (or enum CRYPTO_TYPE = {SYMMETRIC, PUBLIC_PRIVATE}) as a param to this class, based on that value use the correct ALGO and keep those private static internal vars?

Reply:
I think this comment was missed?